modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java

```diff
@@ -348,6 +376,16 @@ public PNGImageWriteParam() {
     private ROI roi;
 
     /**
+     * The NoData range to check nodata, or {@code null} if none.
+     */
+    private Range nodata;
+
+    /**
+     * Array of values used for indicating the background values
+     */
+    private double[] destNoData;
```

Comment:
How comes the input nodata has to be a range (compact set of values) and the destination ones can be a scattered array instead?